gammapy/datasets/map.py

```diff
@@ -489,10 +489,9 @@ def npred_background(self):
 
     def _background_parameters_changed(self):
         values = self.background_model.parameters.value
-        # TODO: possibly allow for a tolerance here?
-        # changed = not np.allclose(self._background_parameters_cached, values, rtol=1e-3)
-
-        changed = ~np.all(self._background_parameters_cached == values)
+        changed = not np.allclose(
+            self._background_parameters_cached, values, rtol=1e-15
+        )
```

Comment:
I would just remove the TODO and keep the previous behaviour because if the parameters did not change the cached values should be exactly the same.